src/libled_internal.h

```diff
@@ -0,0 +1,128 @@
+/*
+ * LED library internal
+ *
+ * Copyright (C) 2022-2022 Red Hat, Inc.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ */
+
+#ifndef _LIB_LED_INTERNAL_INCLUDED_
+#define _LIB_LED_INTERNAL_INCLUDED_
+
+#include "led/libled.h"
+#include "sysfs.h"
+#include "list.h"
+#include "amd_sgpio.h"
+
+#include <linux/limits.h>
+
+struct amd_sgpio_state {
+	int cache_fd;
+	struct cache_entry *cache;
+};
+
+struct configuration {
+	int blink_on_migration;
+	int blink_on_init;
+	int rebuild_blink_on_all;
+	int raid_members_only;
+
+	struct list allowlist;
+	struct list excludelist;
+};
+
+struct led_slot_list_entry {
+	char device[PATH_MAX];
+	char slot[PATH_MAX];
+	enum led_cntrl_type cntrl;
+	enum led_ibpi_pattern ibpi;
+};
+
+struct led_slot_list {
+	struct list slot_list;
+	struct node *iter;
+};
+
+
+struct led_cntrl_list_entry {
+	char path[PATH_MAX];
+	enum led_cntrl_type cntrl_type;
+};
+
+struct led_cntrl_list {
+	struct list cntrl_list;
+	struct node *iter;
+};
+
+struct led_ctx {
+	struct sysfs sys;
+	int log_fd;
+	enum led_log_level_enum log_lvl;
+	led_status_t deferred_error;
+	int dellssd_hw_gen;	/* cached value for dellssd hardware generation */
+	long ipmi_msgid;	/* incrementing message id */
+	struct amd_sgpio_state amd_sgpio;
+	struct configuration config;
+
+	struct led_cntrl_list cl;
+	struct led_slot_list sl;
+};
+
+/**
+ * @brief Add a device path to an allow list.  The library will then only consider this device
+ * for API calls.  Function can be repeated, to add to an internal list.
+ * 
+ * @param[in]	ctx	Library context
+ * @param]in]	path	Device path in allow
+ * @return led_status_t	LED_STATUS_SUCCESS on sucess, else error reason.
+ * 
+ * Note: Mutually exclusive with exclude, allow takes priority.
+ */
+led_status_t device_allow_pattern_add(struct led_ctx *ctx, const char *path);
+
+/**
+ * @brief Add a device path to an exclude list.  The library will then exclude this device
+ * from API calls.  Function can be repeated, to add to an internal list.
+ * 
+ * @param[in]	ctx	Library context 
+ * @param[in]	path	Device path to exclude 
+ * @return led_status_t LED_STATUS_SUCCESS on sucess, else error reason.
+ */
+led_status_t device_exclude_pattern_add(struct led_ctx *ctx, const char *path);
```

Comment:
Shouldn't it be part of libled.h?

Reply:
I can see that you moved it to libled_internal.h recently. What is the reason behind that?
I think that it should be configurable via library.

Reply:
> I can see that you moved it to libled_internal.h recently. What is the reason behind that? I think that it should be configurable via library.

This was based on your comment: https://github.com/intel/ledmon/pull/115#discussion_r1055356788
stating that this was an internal ledmon property.

Reply:
Agree. I have mixed feelings about that as you can see.